controllers/provisioners/eks/eks_test.go

```diff
@@ -92,13 +96,39 @@ func MockKubernetesClientSet() kubeprovider.KubernetesClientSet {
 }
 
 func MockContext(instanceGroup *v1alpha1.InstanceGroup, kube kubeprovider.KubernetesClientSet, w awsprovider.AwsWorker) *EksInstanceGroupContext {
+	var (
+		clusterCa       = "somestring"
+		clusterEndpoint = "foo.amazonaws.com"
+	)
 	input := provisioners.ProvisionerInput{
 		AwsWorker:     w,
 		Kubernetes:    kube,
 		InstanceGroup: instanceGroup,
 		Log:           ctrl.Log.WithName("unit-test").WithName("InstanceGroup"),
 	}
-	return New(input)
+	context := New(input)
+
+	context.DiscoveredState = &DiscoveredState{
+		Provisioned:          false,
+		NodesReady:           false,
+		ClusterNodes:         nil,
```

Comment:
we can probably remove all the fields which are nil as that would be the default.